helix-rest/src/main/java/org/apache/helix/rest/server/ServerContext.java

```diff
@@ -117,9 +121,20 @@ public HelixDataAccessor getDataAccssor(String clusterName) {
     }
   }
 
+  public ZkBaseDataAccessor<ZNRecord> getZkBaseDataAccessor(ZkSerializer serializer) {
+    synchronized (_zkBaseDataAccessorBySerializer) {
+      if (!_zkBaseDataAccessorBySerializer.containsKey(serializer)) {
```

Comment:
The key of this map is the reference to the ZkSerializer, your later code where this method is called always  creates a new ZkSerializer and passes to this  method, that  will always  create  a new  BaseDataAccessor, and  since the basedataAccessor created here never get closed, it will cause  ZkClient leak.

Reply:
+1

I think what @lei-xia meant was that since you create a new ZkSerializer (propertyStoreDataSerializer) every time the endpoint is called, `_zkBaseDataAccessorBySerializer.containsKey(serializer)` will always return false and you'll end up creating a new ZkBaseDataAccessor.

Reply:
@lei-xia @narendly Thanks for catching it! I was thinking that a new baseDataAccessor per each zkSerializer (based on the content of serializer itself). My latest commit addresses the loophole of multiple zkConnections where the propertyStore serializer's hashcode is identical. To answer the question "shall we close the Zkconnection", I don't think the cost is much given the fact current HelixZkClient inside Helix-Rest server is never closed. As long as the helix-rest service is up and running, it keeps a constant connection to the zk cluster.